src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/validappeal/ValidateAppealAboutToSubmitHandler.java

```diff
@@ -57,7 +90,126 @@ public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType,
+                        appeal.getBenefitType().getCode(),
+                        appeal.getMrnDetails().getDwpIssuingOffice());
+
+                callback.getCaseDetails().getCaseData().setDwpRegionalCentre(dwpRegionCentre);
+            }
+
+            String postcode = appealPostcodeHelper.resolvePostcode(appeal.getAppellant());
+            String processingVenue = sscsDataHelper.findProcessingVenue(postcode, appeal.getBenefitType());
+
+            if (isNotBlank(processingVenue)) {
+                callback.getCaseDetails().getCaseData().setProcessingVenue(processingVenue);
+                Optional<CaseManagementLocation> caseManagementLocationOptional = caseManagementLocationService
+                        .retrieveCaseManagementLocation(processingVenue, callback.getCaseDetails().getCaseData().getRegionalProcessingCenter());
+
+                caseManagementLocationOptional.ifPresent(caseManagementLocation ->
+                        callback.getCaseDetails().getCaseData()
+                                .setCaseManagementLocation(caseManagementLocation));
+            }
+            setCaseAccessManagementCategories(appeal, callback);
+        } else {
+            setUnknownCategory(callback);
+        }
+
+        setCaseAccessManagementFields(appeal, callback);
+    }
+
+    private void setCaseAccessManagementCategories(Appeal appeal, Callback<SscsCaseData> callback) {
+        if (caseAccessManagementFeature) {
+            Optional<Benefit> benefit = Benefit.getBenefitOptionalByCode(appeal.getBenefitType().getCode());
+            benefit.ifPresent(
+                    value -> callback.getCaseDetails().getCaseData().getCaseAccessManagementFields()
+                            .setCategories(value));
+        }
+    }
+
+    private void setCaseAccessManagementFields(Appeal appeal, Callback<SscsCaseData> callback) {
+        if (caseAccessManagementFeature) {
+            if (appeal != null && appeal.getAppellant() != null && appeal.getAppellant().getName() != null
+                    && appeal.getAppellant().getName().getFirstName() != null && appeal.getAppellant().getName().getLastName() != null) {
+                callback.getCaseDetails().getCaseData().getCaseAccessManagementFields().setCaseNames(appeal.getAppellant().getName().getFullNameNoTitle());
+            }
+            if (appeal != null && appeal.getBenefitType() != null) {
+                FormType formType = callback.getCaseDetails().getCaseData().getFormType();
+                Optional<Benefit> benefit = Benefit.getBenefitOptionalByCode(appeal.getBenefitType().getCode());
+                String ogdType = isHmrcBenefit(benefit, formType) ? "HMRC" : "DWP";
+                callback.getCaseDetails().getCaseData().getCaseAccessManagementFields().setOgdType(ogdType);
+            }
+        }
+    }
+
+    private boolean isHmrcBenefit(Optional<Benefit> benefit, FormType formType) {
```

Comment:
Is this only for HMRC? If so we won't need to add any validation for new type -> SSCS8 which is used by IBA

Reply:
it looks like it's using this to differentiate between SSCS types and set ogdType.
So will have to check for SSCS8 as well, but what would we set the ogdType to?